app/src/main/java/com/univoice/data_remote/datasourceimpl/UserPreferencesDataSourceImpl.kt

```diff
@@ -0,0 +1,31 @@
+package com.univoice.data_remote.datasourceimpl
+
+import android.content.Context
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.univoice.data.datasource.UserPreferencesDataSource
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+private val Context.dataStore by preferencesDataStore(name = "user_preferences")
+
+class UserPreferencesDataSourceImpl(private val context: Context) : UserPreferencesDataSource {
```

Comment:
P2 : 
```suggestion
class UserPreferencesDataSourceImpl @Inject constructor(
    private val dataStore: DataStore<Preferences>,
) : UserPreferencesDataSource {
```
이렇게 사용할 수 있어요!
제가 심화 스터디에서 부족하지만 구현했던 PR인데 참고해서 비교해보고 업뎃하면 좋을 것 같네요~
https://github.com/NOWSOPT34-Advanced-Study/Team3/pull/15/files